### ISSUE TITLE
How can I implement semi-supervised learning with cnn or lstm for texts classification except self-training?

### ISSUE THREAD

**Issue by Imorton-zd** [fchollet/keras#2249]
Are there some examples in keras to do semi-supervised learning with cnn or lstm for texts classification except self-training? Any opinions would be appreciated!

**Comment by Imorton-zd** [comment]
@braingineer @codekansas @joelthchao Would you give me some suggestions? Thanks!

**Comment by joelthchao** [comment]
Can you provide a more clear direction of your purpose?

**Comment by Imorton-zd** [comment]
@joelthchao I'm sorry that my description is so vague. For example, I have 200 product reviews as training samples, which is labeled with positive or negative, 200 test reviews and a large number of unlabeled reviews about 1600. If I don't care the unlabeled reviews, I directly use the labeled reviews to predict the test samples, the classification accuracy may be a little low. Now if I want to leverage the unlabeled reviews to help improve the classification performance. The easy way is self-training method, which means using 200 labeled samples to predict the unlabeled samples with LSTM or CNN, then mixing the labeled and predicted samples to train a classifier, finally testing the test samples. Of course, it may lead a performance loss if the predictions of unlabeled data are too much wrong. Therefore, what measures can I take to implement semi-supervised learning with cnn or lstm for texts classification?

**Comment by joelthchao** [comment]
Your problem has a relative small training samples and a big unlabeled data, therefore you can try semi-supervised clustering method. Have a look on [GitXiv](http://gitxiv.com/posts/SQGj94ZFTebfAswLF/neural-network-based-clustering-using-pairwise-constraints).

**Comment by braingineer** [comment]
[there is a large literature on this problem in general](https://scholar.google.com/scholar?q=sparse+labels+semi+supervised).

**Comment by Imorton-zd** [comment]
@joelthchao Thanks very much for your suggestion! I have read the page and found it depends on torch, but I have no idea of torch, and the code from torch transforming to keras is too hard. Would you recommend some depending on keras? It will be easy with me!

**Comment by zero76114** [comment]
I have a similar problem. I need to implement semi-supervised learning for named entity recognition in Keras. Any idea for me? many thanks!

**Comment by zero76114** [comment]
@braingineer : As you told it 's really large, so too hard for me. 
In thread How to Implement A Semi-Supervised Model #2380 as mention.  However, I hope more example.
Thanks.
